### PR TITLE
GLWindow: handle middle mouse button press, use for zoom

### PR DESCRIPTION
**Pull request by NeroBurner**
- handle the middle mouse button press
- when the middle mouse button is pressed use the delta along the y axis
  to zoom the view (like the mouse wheel)

fixes https://github.com/CloudCompare/CloudCompare/issues/382

### CODE REVIEW

File: libs/qCC_glWindow/ccGLWindow.cpp [hunk 1]
```diff
@@ -4248,6 +4256,14 @@ void ccGLWindow::mouseMoveEvent(QMouseEvent *event)
 			}
 		}
 	}
+	else if ((event->buttons() & Qt::MiddleButton)) // zoom
+	{
+		//middle button = zooming
+		float pseudo_wheelDelta_deg = -dy;
```
Comment (dgirardeau) [inline libs/qCC_glWindow/ccGLWindow.cpp]: Isn't this producing a warning about type conversion ? Can you maybe use static_cast<float>(-dy)?

**Comment by dgirardeau** [comment]
Thanks for this contribution.

Can you also add a line about this new feature in the CHANGELOG.md file?

File: libs/qCC_glWindow/ccGLWindow.cpp (hunk 1 above)
Comment (NeroBurner) [reply libs/qCC_glWindow/ccGLWindow.cpp]: did not warn for me (qtcreators clang-tidy and clazy warnings), but added it anyways

**Comment by NeroBurner** [comment]
added to changelog

**Pull request merged**